cli/azd/pkg/tools/maven/maven.go

```diff
@@ -138,6 +143,36 @@ func getMavenWrapperPath(projectPath string, rootProjectPath string) (string, er
 	}
 }
 
+// cMavenVersionRegexp captures the version number of maven from the output of "mvn --version"
+//
+// the output of mvn --version looks something like this:
+// Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
+// Maven home: C:\Tools\apache-maven-3.9.1
+// Java version: 17.0.6, vendor: Microsoft, runtime: C:\Program Files\Microsoft\jdk-17.0.6.10-hotspot
+// Default locale: en_US, platform encoding: Cp1252
+// OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"
+var cMavenVersionRegexp = regexp.MustCompile("Apache Maven (.*) \\([0-9a-f]{40}\\)")
```

Comment:
Was thinking about whether the sha-hash length here was stable.. Maybe if git ever change the hash length? Almost wonder if simply matching of the starting parenthesis ends up working better in practice

Reply:
Yeah, I can move to that.  That seems like it will be a little less fragile in the face of changes from the tools.